personio-personnel-data-api-oa3.yaml

```diff
@@ -3828,6 +3969,67 @@ components:
         - end_date
         - half_day_start
         - half_day_end
+    CreateAbsencePeriodRequest:
+      title: Create Absence periods **tracked in hours**
+      type: object
+      properties:
+        employee_id:
+          type: integer
+          description: Employee identifier
+        absence_type_id:
+          type: string
+          format: uuid
+          description: Absence type identifier
+        start_date:
+          type: string
+          format: date
+          description: "Absence start date. Format: yyyy-mm-dd"
+          example: '2020-01-31'
+        end_date:
+          type: string
+          format: date
+          description: "Absence end date. Format: yyyy-mm-dd"
+          example: '2020-01-31'
+        full_day:
+          type: boolean
+          example: false
+          description: Whether it's a full day absence or not, only required if the start_date and end_date are the same (one day absence at most)
+        start_time:
+          type: string
+          format: date
+          description: "Absence start time. Format: HH:ii and only required if the start_date and end_date are the same, and full_day is false (less than one day absence)"
+          example: '03:00'
+        end_time:
+          type: string
+          format: date
+          description: "Absence end time. Format: HH:ii and only required if the start_date and end_date are the same, and full_day is false (less than one day absence)"
+          example: '15:00'
+        half_day_start:
+          type: boolean
+          example: false
+          description: Whether the start date is a half-day off, only required if the start_date and end_date are not the same (more than one day absence)
+        half_day_end:
+          type: boolean
+          example: false
+          description: Whether the end date is a half-day off, only required if the start_date and end_date are not the same (more than one day absence)
+        comment:
+          type: string
+          example: Some Comment
+          description: Optional comment
+        skip_approval:
+          type: boolean
+          example: false
+          description: Optional, default value is true. If set to false, the approval status of the absence request will be "pending" if an approval rule is set for the absence type in Personio. The respective approval flow will be triggered.
+      required:
+        - employee_id
+        - absence_type_id
+        - start_date
+        - end_date
+        - full_day
+        - start_time
```

Comment:
`start_time` and `end_time` are only conditionally required, which means they should not be included here?

Reply:
Is this a question or an info? for me Idk I left it to the code review, if you're sure let me know I will remove it along with `full_day`, `half_day_start `, and `half_day_end `

Reply:
I think this case is still not supported, there is a good suggestion here https://github.com/OAI/OpenAPI-Specification/issues/256#issuecomment-72127524 but still not formally correct, wdyt which way to go?

Reply:
Yes, it's unfortunate that we can't express our validation logic here, but I think it makes sense to only list the properties that are **always** required here.